node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -499,6 +508,7 @@ void converge() {
                 storageMaintainer.cleanupNodeStorage(containerName, nodeSpec);
                 updateNodeRepoWithCurrentAttributes(nodeSpec);
                 nodeRepository.markNodeAvailableForNewAllocation(hostname);
+                expectNodeNotInNodeRepo = true;
```

Comment:
is there a time window where node is in ready before it's removed?

Reply:
No, the HTTP request to mark available for new allocation immediately deletes the node:
https://github.com/vespa-engine/vespa/blob/13b092d7ccb756bad38b7ee26fb781ace108e51c/node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java#L496